src/controllers/AuthLocalController.js

```diff
@@ -0,0 +1,72 @@
+import { User } from '../queries';
+import * as helper from '../helpers';
+import * as validate from '../helpers/validation';
+import status from '../config/status';
+
+/**
+ * A class to handle user local authentication
+ */
+export default class AuthLocalController {
+  /**
+   * @param {object} req
+   * @param {object} res
+   * @return {object} user information & token
+   */
+  static async signup(req, res) {
+    const {
+      firstName, lastName, username, email
+    } = req.body;
+    try {
+      const isUser = await validate.isUser({ email });
+      if (isUser) {
+        return res.status(status.EXIST).send({ error: 'Sorry, this account already exists' });
+      }
+      const hashedPassword = helper.password.hash(req.body.password);
+      const newUser = await User.create({
+        firstName, lastName, username, email, password: hashedPassword
+      });
+      if (!newUser.errors && newUser && Object.keys(newUser).length > 0) {
+        delete newUser.password;
+        return res.status(status.CREATED).send({
+          user: newUser,
+          token: helper.tokenGenerator({ email, role: req.body.isAdmin })
+        });
+      }
+    } catch (error) {
+      return res.status(status.SERVER_ERROR).send({ error });
+    }
+  }
+
+  /**
+   * @description - login user function
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} user token
+   */
+  static async login(req, res) {
+    try {
+      const { email, password } = req.body;
+      const checkUser = await User.findOne({ email });
+      if (Object.keys(checkUser).length > 0) {
+        const comparePassword = helper.password.compare(password, checkUser.password);
+        if (!comparePassword) {
+          return res.status(status.UNAUTHORIZED).send({
+            message: 'The credentials you provided is incorrect'
```

Comment:
create `const` holders for constant values

Reply:
this has been implemented